Marlin/Marlin_main.cpp

```diff
@@ -347,20 +347,31 @@ bool target_direction;
 #endif
 
 #ifdef DELTA
+
+  #define TOWER_1 X_AXIS
+  #define TOWER_2 Y_AXIS
+  #define TOWER_3 Z_AXIS
```

Comment:
Getting away from XYZ is clearly the thing to do. Those describe a (different) coordinate system.

However, We need to remember that there is a distinction between the physical system and the driver controllers.
(That's a part of the HAL concept). Therefore, so that we don't have to go back and separate them in the future, I think that you should keep separate names and match them up with some defines styled as above.  That way, there will be less to change as we transition to a HAL-based description.